python/ray/serve/common.py

```diff
@@ -32,20 +83,130 @@ class DeploymentStatus(str, Enum):
 
 @dataclass
 class DeploymentStatusInfo:
+    name: str
     status: DeploymentStatus
     message: str = ""
 
+    def __eq__(self, other):
+        if not isinstance(other, DeploymentStatusInfo):
+            return False
+
+        return (
+            self.name == other.name
+            and self.status == other.status
+            and self.message == other.message
+        )
+
+    def __str__(self):
+        return (
+            f"Name: {self.name}\n" f"Status: {self.status}\n" f"Message: {self.message}"
+        )
+
     def to_proto(self):
-        return DeploymentStatusInfoProto(status=self.status, message=self.message)
+        return DeploymentStatusInfoProto(
+            name=self.name, status=self.status, message=self.message
+        )
 
     @classmethod
     def from_proto(cls, proto: DeploymentStatusInfoProto):
         return cls(
+            name=proto.name,
             status=DeploymentStatus(DeploymentStatusProto.Name(proto.status)),
             message=proto.message,
         )
 
 
+@dataclass
+class StatusInfo:
```

Comment:
more descriptive name?

Reply:
Any suggestions? How about `ServeStatusInfo`?

Reply:
Resolved in [this thread](https://github.com/ray-project/ray/pull/24839#discussion_r879793626).